examples/webxr_ar_lighting.html

```diff
@@ -0,0 +1,176 @@
+<!DOCTYPE html>
+<html lang="en">
+	<head>
+		<title>three.js ar - cones</title>
+		<meta charset="utf-8">
+		<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no">
+		<link type="text/css" rel="stylesheet" href="main.css">
+	</head>
+	<body>
+
+		<div id="info">
+			<a href="https://threejs.org" target="_blank" rel="noopener">three.js</a> ar - Lighting Estimation<br/>(Chrome Android 88+)
+		</div>
+
+		<script type="module">
+
+			import * as THREE from '../build/three.module.js';
+			import { RGBELoader } from './jsm/loaders/RGBELoader.js';
+			import { ARButton } from './jsm/webxr/ARButton.js';
+			import { XREstimatedLight } from './jsm/webxr/XREstimatedLight.js';
+
+			let camera, scene, renderer;
+			let controller;
+			let defaultEnvironment;
+			let ballGroup;
+
+			init();
+			animate();
+
+			function init() {
+
+				const container = document.createElement( 'div' );
+				document.body.appendChild( container );
+
+				scene = new THREE.Scene();
+
+				camera = new THREE.PerspectiveCamera( 70, window.innerWidth / window.innerHeight, 0.01, 20 );
+
+				const defaultLight = new THREE.HemisphereLight( 0xffffff, 0xbbbbff, 1 );
+				defaultLight.position.set( 0.5, 1, 0.25 );
+				scene.add( defaultLight );
+
+				//
+
+				renderer = new THREE.WebGLRenderer( { antialias: true, alpha: true } );
+				renderer.setPixelRatio( window.devicePixelRatio );
+				renderer.setSize( window.innerWidth, window.innerHeight );
+				renderer.outputEncoding = THREE.sRGBEncoding;
+				renderer.physicallyCorrectLights = true;
+				renderer.xr.enabled = true;
+				container.appendChild( renderer.domElement );
+
+				// Don't add the XREstimatedLight to the scene initially.
+				// It doesn't have any estimated lighting values until an AR session starts.
+
+				const xrLight = new XREstimatedLight( renderer );
+
+				xrLight.addEventListener( 'estimationstart', () => {
+
+					// Swap the default light out for the estimated one one we start getting some estimated values.
+					scene.add(xrLight);
+					scene.remove(defaultLight);
+
+					// The estimated lighting also provides an environment cubemap, which we can apply here.
+					scene.environment = xrLight.environment;
```

Comment:
Agreed that it's nice this is optional. Since high-res reflection with PMREM is so important for PBR, I'm thinking I might skip the estimated environment and use a static PMREM'ed one, but maybe a non-HDR like a jpg, so that most of the intense lighting comes from the directional and SH. I'm not a big fan of SH, but it's probably better than the alternatives.

Reply:
I'd be really interested in seeing the results you can get out of that! There's no question that the low res environment map has limits. Everything tends to look rougher than intended above a certain point. That said if you have a perfectly crisp reflection that doesn't match your environment that will probably stand out too.

I wonder if there's a middle ground where you tone map a high res generic environment with the colors coming out of the low res one. 🤔 Gonna guess that falls into the "More trouble than it's worth" category, though.

Reply:
I'm hoping the SH will tend to provide that tone map. As for the perfectly crisp reflection; no one will notice except on a spherical mirror like you have here. Most real objects are complex enough that you really can't tell what's reflected anyway. All that matters is how sharp the features are, which we need resolution to represent.